store/src/java/com/zimbra/cs/service/mail/ParseMimeMessage.java

```diff
@@ -284,7 +284,7 @@ private static class ParseMessageContext {
                 long mailAttachmentMaxSize = account.getMailAttachmentMaxSize();
                 Config config = Provisioning.getInstance().getConfig();
                 long mtaMaxMsgSize = config.getLongAttr(Provisioning.A_zimbraMtaMaxMessageSize, -1);
-                if (mailAttachmentMaxSize > mtaMaxMsgSize) {
+                if (mailAttachmentMaxSize > mtaMaxMsgSize || mailAttachmentMaxSize == 0) {
```

Comment:
@viditZim could you check if the request fails if the mailAttachmentMaxSize & mtaMaxMsgSize is not set and maxsize is -1

Reply:
its max value then

Reply:
In that case, the limit will be unlimited because we're setting as Long.MAX_VALUE for negative.

https://github.com/Zimbra/zm-mailbox/blob/36f7286f28527de70341b10f614903ff936e406a/store/src/java/com/zimbra/cs/service/mail/ParseMimeMessage.java#L295